### DISCUSSION TITLE
Is it possible to run Prisma Studio on heroku?

### DISCUSSION THREAD

**Discussion by felinto-dev** [prisma/prisma#9373] [Q&A]
Is it possible to run Prisma Studio on heroku?

How could I do it?

**Comment by sdnts** [comment]
Hey @felinto-dev, for Heroku you should be able to run ‘npx prisma studio’ directly on the port you want, right?

If not, there is an undocumented programmatic API, but I would advise against using it generally, because it isn’t public API, so it might change without notice. You should be able to refer to the code we have for the Prisma CLI: https://github.com/prisma/prisma/blob/main/packages/cli/src/Studio.ts#L114

@prisma/studio-server is the NPM package that we use (and maintain, but there might be breaking changes between non-major versions, since it is not a public API)

Studio is a regular client + server app, which means there is a frontend (which is the UI you see) and a backend (which executes Prisma Client requests and talks to your database). I think you might be able to copy over most of the implementation front the CLI.  Some things to take care of:
1. You need a schema on disk, but I think that shouldn’t be a problem since you likely will have access to your repository on disk on Heroku
2. The staticAssetDir is the directory that contains Studio’s frontend assets. On the CLI, we copy them out, but you should be able to use ‘require.resolve(“@prisma/studio”)’ instead. @prisma/studio is the NPM package for Studio’s frontend. 
3. You’ll need @prisma/sdk as a dependency since it is marked as a peerDependency for studio-server

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
